sdk-extensions/aws/src/main/java/io/opentelemetry/sdk/extension/aws/trace/SamplingRuleApplier.java

```diff
@@ -0,0 +1,248 @@
+          rule.getAttributes().entrySet().stream()
+              .collect(Collectors.toMap(Map.Entry::getKey, e -> toMatcher(e.getValue())));
+    }
+
+    urlPathMatcher = toMatcher(rule.getUrlPath());
+    serviceNameMatcher = toMatcher(rule.getServiceName());
+    httpMethodMatcher = toMatcher(rule.getHttpMethod());
+    hostMatcher = toMatcher(rule.getHost());
+    serviceTypeMatcher = toMatcher(rule.getServiceType());
+    resourceArnMatcher = toMatcher(rule.getResourceArn());
+  }
+
+  boolean matches(String name, Attributes attributes, Resource resource) {
+    int matchedAttributes = 0;
+    String httpTarget = null;
+    String httpMethod = null;
+    String host = null;
+
+    for (Map.Entry<AttributeKey<?>, Object> entry : attributes.asMap().entrySet()) {
+      if (entry.getKey().equals(SemanticAttributes.HTTP_TARGET)) {
+        httpTarget = (String) entry.getValue();
+      } else if (entry.getKey().equals(SemanticAttributes.HTTP_METHOD)) {
+        httpMethod = (String) entry.getValue();
+      } else if (entry.getKey().equals(SemanticAttributes.HTTP_HOST)) {
+        host = (String) entry.getValue();
+      }
+
+      Matcher matcher = attributeMatchers.get(entry.getKey().getKey());
+      if (matcher == null) {
+        continue;
+      }
+      if (matcher.matches(entry.getValue().toString())) {
+        matchedAttributes++;
+      } else {
+        return false;
+      }
+    }
+    // All attributes in the matched attributes must have been present in the span to be a match.
+    if (matchedAttributes != attributeMatchers.size()) {
+      return false;
+    }
+
+    return urlPathMatcher.matches(httpTarget)
+        && serviceNameMatcher.matches(name)
+        && httpMethodMatcher.matches(httpMethod)
+        && hostMatcher.matches(host)
+        && serviceTypeMatcher.matches(getServiceType(resource))
+        && resourceArnMatcher.matches(getArn(attributes, resource));
+  }
+
+  SamplingResult shouldSample(
+      Context parentContext,
+      String traceId,
+      String name,
+      SpanKind spanKind,
+      Attributes attributes,
+      List<LinkData> parentLinks) {
+    return fixedRateSampler.shouldSample(
+        parentContext, traceId, name, spanKind, attributes, parentLinks);
+  }
+
+  @Nullable
+  private static String getArn(Attributes attributes, Resource resource) {
+    String arn = resource.getAttributes().get(ResourceAttributes.AWS_ECS_CONTAINER_ARN);
+    if (arn != null) {
+      return arn;
+    }
+    String cloudPlatform = resource.getAttributes().get(ResourceAttributes.CLOUD_PLATFORM);
+    if (cloudPlatform == null
+        || !cloudPlatform.equals(ResourceAttributes.CloudPlatformValues.AWS_LAMBDA)) {
+      return null;
+    }
+    arn = resource.getAttributes().get(ResourceAttributes.FAAS_ID);
+    if (arn != null) {
+      return arn;
+    }
+    return attributes.get(ResourceAttributes.FAAS_ID);
+  }
+
+  @Nullable
+  private static String getServiceType(Resource resource) {
```

Comment:
A bit of theoretical question, but as this does not cover whole enum, wouldn't map be actually faster here?

Reply:
Would need to bench to confirm but Map sounds good. I think the key point is that hashcodes are cached for Strings in Java so in practice should only be computed around once per app.